src/handlers.cpp

```diff
@@ -1578,29 +1640,120 @@ void ImpuRegDataTask::on_sar_response(Diameter::Message& rsp)
       {
         TRC_DEBUG("Getting User-Data from SAA for cache");
         saa.user_data(_xml);
-        put_in_cache();
+        caching = true;
```

Comment:
Happy with the rearrangement, but I do have one question about your suggested commit.  You have saa.user_data(_xml) being called unconditionally (even if its a dereg or auth failure).  Is this functional change safe?